### ISSUE TITLE
Swagger UI 500 InvalidUrlResource but app-docs works

### ISSUE THREAD

**Issue by pdewilde** [springfox/springfox#2676]
I have tried a lot of the solutions provided in similar threads (#776, #2396, #1615) but am still having problems. I get a 500 status when I try to access swagger-ui, even though swagger seems to be working. I made a new issue since other people did not seem to have this same 500 error. I was wondering if anyone had any ideas on what I should try.

http://localhost:8088/myproject/v2/api-docs works and provides the proper json
http://localhost:8088/myproject/swagger-resources/configuration/ui  also comes up, showing that ui is properly pulled into the project.

http://localhost:8088/myproject/swagger-ui.html returns a 500 with a blank body.

I am running Spring Boot v1.3.6.RELEASE, Spring v4.3.3.RELEASE (I know they are older, could that be a problem?)
For springfox swagger2 and swagger-ui I am running 2.9.2.

Things I have tried:

- Remove @EnableMvc annotation
- Remove a mapping `RequestMapping(value="{versionId}/version", method=RequestMethod.GET)` or any other mappings with variables which may conflict
- Add a whitelist in my Spring Security Configuration for any applicable pahts
- Explicitly put in path for my Docket

When I try to make a request to swagger-ui, this shows up in my log:
```
2018-09-11 12:42:32,863  INFO qtp345780740-21 InvalidUrlResource  - Invalid request - /swagger-ui.html
2018-09-11 12:42:32,863 ERROR qtp345780740-21 BaseJsonResource  - Invalid request - /swagger-ui.html
2018-09-11 12:42:32,864  WARN qtp345780740-21 ServletHandler  - 
org.springframework.web.util.NestedServletException: Request processing failed; nested exception is com.openmarket.service.rest.WebserviceException: Invalid request - /swagger-ui.html
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:982)
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:861)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:812)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1669)
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:224)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:207)
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:176)
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:346)
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:262)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:197)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:577)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
	at org.eclipse.jetty.server.Server.handle(Server.java:499)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:311)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257)
	at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:544)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.openmarket.service.rest.WebserviceException: Invalid request - /swagger-ui.html
	at com.openmarket.service.rest.resource.InvalidUrlResource.handleInvalidUrl(InvalidUrlResource.java:25)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:221)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:136)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:114)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:827)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:738)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:85)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:963)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:897)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)
	... 33 more
2018-09-11 12:42:32,865  WARN qtp345780740-21 ErrorHandler  - EXCEPTION 
javax.servlet.ServletException: Circular view path [error]: would dispatch back to the current handler URL [/myproject/error] again. Check your ViewResolver setup! (Hint: This may be the result of an unspecified view, due to default view name generation.)
	at org.springframework.web.servlet.view.InternalResourceView.prepareForRendering(InternalResourceView.java:205)
	at org.springframework.web.servlet.view.InternalResourceView.renderMergedOutputModel(InternalResourceView.java:145)
	at org.springframework.web.servlet.view.AbstractView.render(AbstractView.java:303)
	at org.springframework.web.servlet.DispatcherServlet.render(DispatcherServlet.java:1257)
	at org.springframework.web.servlet.DispatcherServlet.processDispatchResult(DispatcherServlet.java:1037)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:980)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:897)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:861)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:812)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:587)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:595)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:191)
	at org.eclipse.jetty.server.Dispatcher.error(Dispatcher.java:77)
	at org.eclipse.jetty.server.handler.ErrorHandler.handle(ErrorHandler.java:92)
	at org.springframework.boot.context.embedded.jetty.JettyEmbeddedErrorHandler.handle(JettyEmbeddedErrorHandler.java:55)
	at org.eclipse.jetty.server.Response.sendError(Response.java:597)
	at org.eclipse.jetty.server.Response.sendError(Response.java:544)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:647)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:577)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
	at org.eclipse.jetty.server.Server.handle(Server.java:499)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:311)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257)
	at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:544)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by dilipkrish** [comment]
@pdewilde What is happening here is that there is multiple paths to resolve swagger-ui.html. Dont know your current configuration but I'll try to illustrate the problem with an example.

Sounds like you have a request mapping call it ControllerX

ControllerX seems to have a request mapping (method or controller itself  like `@RequestMapping("/{id}")`. So what is happening is that `/swagger-ui.html` resolves to that endpoint, and looks like that incorrect endpoint resolves it in such a way that it creates a circular view resolution (recursive).

Hope that makes sense.

**Comment by pdewilde** [comment]
Hi @dilipkrish I tried settings my Docket bean to have 
```
    @Bean
    public Docket api() {
        return new Docket(DocumentationType.SWAGGER_2)
                .select()
                    .apis(RequestHandlerSelectors.any())
                    .paths(PathSelectors.any())
                    .build()
                .pathMapping("/foobar/")
                ;
    }
```
but trying to access http://localhost:8088/myproject/foobar/swagger-ui.html
resulted in the same error.
When I look at that log, I thought it was the error page `myproject/error` which had the circular view path, but the cause for the error was the invalid request. Is that right, or is the root cause likely some circular view path?

**Comment by diegotissoni** [comment]
> @pdewilde What is happening here is that there is multiple paths to resolve swagger-ui.html. Dont know your current configuration but I'll try to illustrate the problem with an example.
> 
> Sounds like you have a request mapping call it ControllerX
> 
> ControllerX seems to have a request mapping (method or controller itself like `@RequestMapping("/{id}")`. So what is happening is that `/swagger-ui.html` resolves to that endpoint, and looks like that incorrect endpoint resolves it in such a way that it creates a circular view resolution (recursive).
> 
> Hope that makes sense.

I have this problem, the swagger-ui endpoint resolves to another endpoint from my app. How can I solve this?

**Comment by pdewilde** [comment]
One layer of the onion is peeled off. I had a controller that was a ExceptionHandler for WebserviceException. When I removed the `@RequestMapping` for my `handleInvalidUrl` method, my error changed from a 500 to a 404, and the InvalidMapping error goes away, but I still get that circular view dependency error.

**Comment by pdewilde** [comment]
Ok, I got a fix! I had tried removing the @EnableWebMvc annotation, but not in combination with removing the @RequestMapping for my `handleInvalidUrl()` in the Controller that had an ExceptionHandler. The combination of these two changes (Removing @EnableWebMvc in my Configuration, and removing the mapping for my `handleInvalidUrl()` fixed the issue, and now the UI comes right up. Thank you @dilipkrish for all your work supporting the SpringFox community. You have been in every issue thread I looked at to get help.
